client/src/modules/reports/generate/visit_report/visit_report.config.js

```diff
@@ -0,0 +1,76 @@
+angular.module('bhima.controllers')
+  .controller('visit_reportController', VisitReportController);
+
+VisitReportController.$inject = [
+  '$sce', 'NotifyService', 'BaseReportService', 'AppCache', 'reportData', '$state',
+  'LanguageService', 'moment',
+];
+
+function VisitReportController(
+  $sce, Notify, SavedReports, AppCache,
+  reportData, $state, Languages
+) {
+  const vm = this;
+  const cache = new AppCache('configure_visit_report');
+  const reportUrl = 'reports/visits';
+
+  vm.previewGenerated = false;
+  vm.dateFrom = new Date();
+  vm.dateTo = new Date();
+
+  // check cached configuration
+  checkCachedConfiguration();
+
+  vm.clear = function clear(key) {
+    delete vm[key];
+  };
+
+  vm.clearPreview = function clearPreview() {
+    vm.previewGenerated = false;
+    vm.previewResult = null;
+  };
+
+  vm.preview = function preview(form) {
+    if (form.$invalid) { return 0; }
+
+    const params = {
+      dateFrom : vm.dateFrom,
+      dateTo : vm.dateTo,
+    };
+
+    // update cached configuration
+    cache.reportDetails = angular.copy(params);
+
+    const options = {
+      params,
```

Comment:
Why not just use
```js
angular.extend(options, params);
```
instead of putting the `params` as an object you then have to `JSON.parse()` on the server?